cmd/crossplane/core/core.go

```diff
@@ -50,11 +54,22 @@ func FromKingpin(cmd *kingpin.CmdClause) *Command {
 	cmd.Flag("cache-dir", "Directory used for caching package images.").Short('c').Default("/cache").OverrideDefaultFromEnvar("CACHE_DIR").ExistingDirVar(&c.CacheDir)
 	cmd.Flag("sync", "Controller manager sync period duration such as 300ms, 1.5h or 2h45m").Short('s').Default("1h").DurationVar(&c.Sync)
 	cmd.Flag("leader-election", "Use leader election for the conroller manager.").Short('l').Default("false").OverrideDefaultFromEnvar("LEADER_ELECTION").BoolVar(&c.LeaderElection)
+	cmd.Flag("provider", "Pre-install a Provider by giving its image URI. This argument can be repeated.").StringsVar(&c.Providers)
+	cmd.Flag("configuration", "Pre-install a Configuration by giving its image URI. This argument can be repeated.").StringsVar(&c.Configurations)
```

Comment:
Is there any value in making this a separate subcommand? This way folks could decouple initialising Crossplane from actually running Crossplane if they wanted to? I can't think of many use cases for doing so but it "feels" a little cleaner to me to keep them separate.

Reply:
You're saying that we'd have to make two calls right? So, `crossplane initialize && crossplane start` in `Deployment` YAML?

For context, I added these flags just to handle pre-install mechanism we have in the Helm chart. So, they need to be part of the arguments in `Deployment` that starts Crossplane.

Reply:
Ah, right. Probably not the best place to drop my comment but yes I was thinking `crossplane initialize` and `crossplane [core]` would be separate commands. That way we could either chain them in the one container or possibly let an init container handle the initialization.

Separately, is the idea behind moving the provider and config install logic into Crossplane core that we need the Lock to exist before we do that? Could the Helm chart continue to handle installing them?

Reply:
> Probably not the best place to drop my comment but yes I was thinking crossplane initialize and crossplane [core] would be separate commands.

If we do that, we'd probably need `crossplane core initialize` and `crossplane rbac initialize` separately because `rbac` can start only if some of the CRDs exist, so it needs an initialization period.

Separating the initialization from core would help local debugging as you would not want to run the initialization logic that looks to a certain path for CRDs, that is `/crds`. Also once we have the separate command, I think it makes sense to have an init container for a more proper readiness signaling. I'll do that in this PR.

> Separately, is the idea behind moving the provider and config install logic into Crossplane core that we need the Lock to exist before we do that? Could the Helm chart continue to handle installing them?

Not sure about `Lock`, but for Helm to be able to create `Provider` and `Configuration` objects, their CRDs need to present. I guess we could keep these two CRDs in the Helm chart and still update them in initializer but I'd not want to manage CRDs in two different places. I think if we go with your suggestion of separating `initialize` command, these pre-install commands will make more sense to have as flag.